tools/cudaq-quake/cudaq-quake.cpp

```diff
@@ -448,8 +453,15 @@ int main(int argc, char **argv) {
     }
 
     // Running the verifier to make it easier to track down errors.
+    // The canonicalizer and symbol DCE passes cleanup unused artifacts
+    // generated by the bridge.
     mlir::PassManager pm(&context);
     pm.addPass(std::make_unique<cudaq::VerifierPass>());
+    if (!noSimplify) {
+      pm.addPass(mlir::createCanonicalizerPass());
+      // FIXME: Workaround bug in regenerating the CPU side kernel entry.
```

Comment:
What exactly is the bug here? We haven't generated the new entrypoint function yet.

Reply:
GenKernelExecution is leaning on the CPU side (C++) entry points being declared by the bridge. There is code (actually the same routine) to figure it out if the declaration is missing, but it guesses wrong in 2 of the regression tests.

It is on my list of things to figure out in a follow-up.